app/screens/home/search/team_picker_icon/search_team_slideup.tsx

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+
+import TeamList from '@components/team_sidebar/add_team/team_list';
+import {useIsTablet} from '@hooks/device';
+import BottomSheetContent from '@screens/bottom_sheet/content';
+import {dismissBottomSheet} from '@screens/navigation';
+
+import type TeamModel from '@typings/database/models/servers/team';
+
+type Props = {
+    teams: TeamModel[];
+    teamId: string;
+    setTeamId: (teamId: string) => void;
+    title: string;
+}
+
+export default function SelectTeamSlideUp({teams, title, setTeamId, teamId}: Props) {
+    const testID = 'team_sidebar.add_team_slide_up';
+    const isTablet = useIsTablet();
+    const showTitle = !isTablet && Boolean(teams.length);
+
+    const onPress = useCallback((newTeamId: string) => {
+        setTeamId(newTeamId);
+        dismissBottomSheet();
```

Comment:
Perhaps you need to `await dismissBottomSheet();` first and then call `setTeamId(newTeamId)`

Reply:
If I `await dismissBottomSheet()` first, the user sees two things 

1) The team icon change 
2) the recent changes section flicker from the old team values to the new.  

As it currently is, the change the team, and see the updated underlying data after the dismissal, as it has already changed via `setTeamId`